### PR TITLE
detail customer

### PR DESCRIPTION
**Pull request by chvister**

### CODE REVIEW

File: src/views/test/GeneratedCustomersTable.tsx [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+        { field: "avatarUrl", flex: 1, type: "string", valueFormatter: ({ value }) => value, renderHeader: (params: GridColParams) => (<FormattedMessage id="Customer" defaultMessage="avatarUrl" />) },
+        { field: "createdAt", flex: 1, type: "date", valueFormatter: ({ value }) => intl.formatDate(value), renderHeader: (params: GridColParams) => (<FormattedMessage id="Customer" defaultMessage="createdAt" />) },
+        { field: "email", flex: 1, type: "string", valueFormatter: ({ value }) => value, renderHeader: (params: GridColParams) => (<FormattedMessage id="Customer" defaultMessage="email" />) },
+        { field: "id", flex: 1, type: "string", valueFormatter: ({ value }) => value, renderHeader: (params: GridColParams) => (<FormattedMessage id="Customer" defaultMessage="id" />) },
+        { field: "name", flex: 1, type: "string", valueFormatter: ({ value }) => value, renderHeader: (params: GridColParams) => (<FormattedMessage id="Customer" defaultMessage="name" />) },
+    ];
+    return (<div style={{ height: "400px", width: "100%" }}><DataGrid onRowClick={() => window.location.href = 'generated-customer-detail'} columns={columns} rows={customers} /></div>);
```
Comment (jozef-slezak) [inline src/views/test/GeneratedCustomersTable.tsx:15]: @chvister, is there any react-router construction to navigate to different page?

Comment (jozef-slezak) [inline src/views/test/GeneratedCustomersTable.tsx:15]: toto (preklik na detail zakaznika) vraj nefunguje ked je nasadene na vercel

Comment (jozef-slezak) [reply src/views/test/GeneratedCustomersTable.tsx:15]: `useHistory()` https://ui.dev/react-router-v5-programmatically-navigate/

Comment (mecirmartin) [reply src/views/test/GeneratedCustomersTable.tsx:15]: `useHistory()` bude najjednoduchsi sposob ako nahradit to prepisanie `window.location.href`

Comment (chvister) [reply src/views/test/GeneratedCustomersTable.tsx]: kedze pouzivame tuto verziu "react-router-dom": "^6.0.0-beta.0", tak som tam nasiel funkciu` useNavigate()`, ktora funguje podobne ako` useHistory()`

File: src/views/test/types.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export interface Customer {
+export interface ICustomer {
```
Comment (jozef-slezak) [inline src/views/test/types.ts:1]: Nepouzivajme to zastaralu konvenciu - nazov interface s prefixom `I`

File: src/views/customer/CustomerListView/UpdateCustomer.tsx [hunk 3]
```diff
@@ -8,23 +8,22 @@ import {
...
-    textfield: {
-      marginTop: theme.spacing(2),
-    },
-    button: {
-      margin: theme.spacing(1),
-    }
-  }),
-);
+
+const useStyles = makeStyles((theme: any) => ({
```
Comment (jozef-slezak) [inline src/views/customer/CustomerListView/UpdateCustomer.tsx:12]: Please, do not use any type!

**Pull request merged**